static/app/views/releases/releaseBubbles/useReleaseBubbles.tsx

```diff
@@ -446,50 +349,256 @@ export function useReleaseBubbles({
+      const handleGlobalOut = () => {
+        if (!echartsInstance) {
+          return;
+        }
 
-  const buckets =
-    (hasReleaseBubbles &&
-      releases?.length &&
-      minTime &&
-      maxTime &&
-      createReleaseBuckets({
-        minTime,
-        maxTime,
-        finalTime: releasesMaxTime,
-        releases,
-        desiredBuckets,
-      })) ||
-    [];
+        const series = echartsInstance.getOption().series;
+        const seriesIndex = series.findIndex((s: Series) => s.id === BUBBLE_SERIES_ID);
+        // We could find and include a `dataIndex` to be specific about which
+        // bubble to "downplay", but I think it's ok to downplay everything
+        echartsInstance.dispatchAction({
+          type: 'downplay',
+          seriesIndex,
+        });
+      };
+
+      const handleLegendSelectChanged = (params: LegendSelectChangedParams) => {
+        if (params.name !== 'Releases' || !('Releases' in params.selected)) {
+          return;
+        }
+        const selected = params.selected.Releases;
+
+        // If `legendSelected` is defined, this hook will assume that the
+        // selected state is "controlled" by the calling component (e.g. it
+        // implements its own event handler and keeps its own legend-selected
+        // state). The hook will return the updated chart options accordingly.
+        if (legendSelected !== undefined) {
+          return;
+        }
+        // Callback for when Releases legend status changes -- we want to
+        // adjust the xAxis/grid accordingly when Releases are visible or
+        // not
+        echartsInstance.setOption({
+          xAxis: selected ? releaseBubbleXAxis : defaultBubbleXAxis,
+          grid: selected ? releaseBubbleGrid : defaultBubbleGrid,
+        });
+
+        trackLegend(params);
+      };
+
+      if (echartsInstance) {
+        /**
+         * MouseListeners for echarts. This includes drawing a highlighted area on the
+         * main chart when a release bubble is hovered over.
+         *
+         * Attach directly to instance to avoid collisions with React props
+         */
+        echartsInstance.on('click', handleSeriesClick);
+        echartsInstance.on('mouseover', handleMouseOver);
+        echartsInstance.on('mouseout', handleMouseOut);
+        echartsInstance.on('globalout', handleGlobalOut);
+        echartsInstance.on('legendselectchanged', handleLegendSelectChanged);
+        echartsInstance.getZr().on('mousemove', handleMouseMove);
+      }
+
+      return () => {
+        if (!echartsInstance) {
+          return;
+        }
+
+        echartsInstance.off('click', handleSeriesClick);
+        echartsInstance.off('mouseover', handleMouseOver);
+        echartsInstance.off('mouseout', handleMouseOut);
+        echartsInstance.off('globalout', handleGlobalOut);
+        echartsInstance.off('legendselectchanged', handleLegendSelectChanged);
+        echartsInstance.getZr().off('mousemove', handleMouseMove);
+      };
```

Comment:
Added these cleanups of event handlers since w/ react19 ref callbacks can return cleanup fns.